### PR TITLE
Groundwork

### PR DESCRIPTION
**Pull request by nicpottier**
This is laying the groundwork for introducing counts on ContactGroups..

Just adding the 'count' field on ContactGroup as well as cleaning up some of the Contact fields. (renaming ```status``` to ```is_failed``` and ```is_archived``` to ```is_blocked```)

It occurs to me now that we use 'credits' and 'used' on TopUp to deal with the counts of total and used credits, so maybe ```count``` is the wrong name? Opinions?

### CODE REVIEW

File: temba/msgs/views.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def clean(self):
         resend_allowed = self.user.get_org_group().permissions.filter(codename="broadcast_send")
 
 
-        if action in ['label', 'unlabel', 'archive', 'restore'] and not update_allowed:
+        if action in ['label', 'unlabel', 'archive', 'restore', 'archive'] and not update_allowed:
```
Comment (norkans7) [inline temba/msgs/views.py]: duplicated 'archive' while you meant to add 'block'

Comment (nicpottier) [inline temba/msgs/views.py]: Oh man, good eyes.. I'll see if I can write a unit test to break this too.

File: temba/contacts/models.py [hunk 2]
```diff
@@ -1242,15 +1237,12 @@ class Meta:
 
 class ContactGroup(TembaModel, SmartModel):
     name = models.CharField(verbose_name=_("Name"), max_length=64, help_text=_("The name for this contact group"))
-
     contacts = models.ManyToManyField(Contact, verbose_name=_("Contacts"), related_name='groups')
-
+    count = models.IntegerField(default=0,
+                                verbose_name=_("Count"), help_text=_("The number of contacts in this group"))
```
Comment (ericnewcomer) [inline temba/contacts/models.py]: I agree ```contact_group.count``` feels a little weird. I don't particularly like ```member_count``` either which is the only other option I came up with. I think ```count``` is propbabyl fine.

**Comment by ericnewcomer** [comment]
All pretty straightforward from these eyes.

**Comment by nicpottier** [comment]
Looking to push this today, can I get another glance at this. (final group counts change will go out afterwards, this is step 1)

**Comment by ericnewcomer** [comment]
reyarp

**Pull request merged**